msprime/_msprimemodule.c

```diff
@@ -1784,7 +1784,8 @@ Simulator_init(Simulator *self, PyObject *args, PyObject *kwds)
     Py_ssize_t num_populations = 1;
     int store_migrations = false;
     int store_full_arg = false;
-    int store_unary = false;
+    int additional_nodes = 0;
```

Comment:
Annoyingly, the Python C API doesn't support fixed width types like ``uint32_t``. So, the simplest thing to do then is to use a type that's guaranteed to be at least this size, and  to cast to uint32_t later. So,
```
unsigned long additional_nodes
```

We use ``k`` for unsigned long in the [Python C API](https://docs.python.org/3/c-api/arg.html)